### PR TITLE
Add boardsource/lulu RP2040 support

### PR DESCRIPTION
**Pull request by waffle87**
## Description

Adds support for the PCBs sold in group buy. 

## Types of Changes

- [x] Keyboard (addition or update)

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/boardsource/lulu/readme.md [hunk 1]
```diff
@@ -12,11 +12,18 @@ The lulu is what the ergo community has needed for a long time, a high-end aesth
...
-    make boardsource/lulu:default:flash
+    make boardsource/lulu/rp2040:default:flash
+    make boardsource/lulu/avr:default:flash
+
+
+Compile `rp2040` firmware if you purchased PCB in lulu group buy (integrated microcontroller).
+Compile `avr` firmware if your PCB uses a drop-in microcontroller like Pro Micro.
+
 
 Reset keyboard by holding down top left key while you plug in the usb cable. Or by pushing the reset switch on the pcb.
```
Comment (keyboard-magpie) [inline keyboards/boardsource/lulu/readme.md:28]:
```suggestion
Enter bootloader by holding down the upper outer key while you plug in the usb cable, or by pushing the reset switch on the pcb.
```

**Comment by keyboard-magpie** [comment]
Oh. Still failing lint...

File: keyboards/boardsource/lulu/avr/config.h [hunk 2]
```diff
@@ -0,0 +1,4 @@
+// Copyright 2022 jack (@waffle87)
+// SPDX-License-Identifier: GPL-2.0-or-later
+#pragma once
+#define RGB_DI_PIN D3
```
Comment (drashna) [inline keyboards/boardsource/lulu/avr/config.h:4]: This could be added to rgblight->pin in the info.json file.

**Review by drashna — APPROVED** [review]

**Review by drashna — COMMENTED** [review]
> KeyError: 'matrix_size'

Specific lint error.

**Review by drashna — APPROVED** [review]

**Review by keyboard-magpie — APPROVED** [review]

**Pull request merged**